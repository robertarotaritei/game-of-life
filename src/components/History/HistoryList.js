import React from 'react';
import Game from '../GameOfLifeGrid/Game'

class HistoryList extends React.Component {

    render() {
        return (
            <div>
                {this.props.games.map(currentGame => (
                    <Game key={currentGame.id} game={currentGame} history={true} />
                ))}
                <div style={{ paddingBottom: 20 }} />
            </div>
        );
    }
}

export default HistoryList;