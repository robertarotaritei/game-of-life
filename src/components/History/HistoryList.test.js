import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import HistoryList from './HistoryList';
import { shallow } from "enzyme";

const games = [
  {
    id: 1,
    initialState: '[[true, true],[false, false]]'
  }
];

test("HistoryList renders without crashing", () => {
  shallow(<HistoryList games={games} />);
});

it('HistoryList renders correctly', () => {
  const tree = renderer
    .create(<HistoryList games={games} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});