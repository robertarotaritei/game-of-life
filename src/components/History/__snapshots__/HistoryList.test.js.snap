// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "marginTop": "30px",
      }
    }
  >
    <div>
      <div>
        <p>
          's Game
        </p>
        <div
          style={
            Object {
              "display": "flex",
              "paddingBottom": "10px",
              "verticalAlign": "middle",
            }
          }
        >
          <button
            className="btnIcon"
            data-testid="play"
            onClick={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8 5v14l11-7z"
              />
            </svg>
          </button>
          <button
            className="btnIcon"
            data-testid="speed"
            onClick={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20.38 8.57l-1.23 1.85a8 8 0 01-.22 7.58H5.07A8 8 0 0115.58 6.85l1.85-1.23A10 10 0 003.35 19a2 2 0 001.72 1h13.85a2 2 0 001.74-1 10 10 0 00-.27-10.44zm-9.79 6.84a2 2 0 002.83 0l5.66-8.49-8.49 5.66a2 2 0 000 2.83z"
              />
            </svg>
          </button>
          <button
            className="btnIcon"
            data-testid="back"
            onClick={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z"
              />
            </svg>
          </button>
          <div>
            <p
              className="textAnimationSave"
              style={
                Object {
                  "color": "red",
                  "paddingBottom": "10px",
                }
              }
            >
                
            </p>
          </div>
        </div>
        <div
          style={
            Object {
              "display": "grid",
              "gridTemplateColumns": "repeat(80, 16px)",
            }
          }
        >
          <div
            className="cell"
            data-testid="0-0"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "backgroundColor": "#009ECE",
              }
            }
          />
          <div
            className="cell"
            data-testid="0-1"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "backgroundColor": "#9BE8FF",
              }
            }
          />
          <div
            className="cell"
            data-testid="1-0"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "backgroundColor": undefined,
              }
            }
          />
          <div
            className="cell"
            data-testid="1-1"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "backgroundColor": undefined,
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "paddingBottom": 20,
      }
    }
  />
</div>
`;
